src/classes/Warehouse.java

```diff
@@ -196,5 +201,57 @@ private void scnRemoveCart() {
 
         carts.remove(cartsName.get(intChoice));
     }
+
+    public void scnResearch() {
+        Scanner scanner = new Scanner(System.in);

```

Comment:
Fare in modo che parti di questo metodo vengano estratte, mantenendo qui solo la lettura dell'input da tastiera.

Reply:
Compiuto nell'ultimo aggiornamento del branch